src/task.c

```diff
@@ -220,6 +220,12 @@ JL_DLLEXPORT void *jl_task_stack_buffer(jl_task_t *task, size_t *size, int *tid)
     return (void *)((char *)task->stkbuf + off);
 }
 
+static void record_backtrace(jl_ptls_t ptls) JL_NOTSAFEPOINT
+{
+    // storing bt_size in ptls ensures roots in bt_data will be found
+    ptls->bt_size = rec_backtrace(ptls->bt_data, JL_MAX_BT_SIZE);
```

Comment:
I agree this is nice to have factored out (especially for having a place to put the note about rooting). I only removed it for symmetry with the equivalent code which had removed it in `partr.c`. We should probably just call `record_backtrace` there as well.